libs/server/Resp/Objects/SetCommands.cs

```diff
@@ -505,5 +506,68 @@ private unsafe bool SetPop<TGarnetApi>(int count, byte* ptr, ref TGarnetApi stor
             return true;
         }
 
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <typeparam name="TGarnetApi"></typeparam>
+        /// <param name="count"></param>
+        /// <param name="ptr"></param>
+        /// <param name="storageApi"></param>
+        /// <returns></returns>
+        private unsafe bool SetMove<TGarnetApi>(int count, byte* ptr, ref TGarnetApi storageApi)
+            where TGarnetApi : IGarnetApi
+        {
+            if (count != 3)
+            {
+                setItemsDoneCount = setOpsCount = 0;
+                return AbortWithWrongNumberOfArguments("SMOVE", count);
+            }
+            else
+            {
+                ArgSlice sourceKey = default;
+                ArgSlice destinationKey = default;
+                ArgSlice sourceMember = default;
+
+                // Get the source key 
+                if (!RespReadUtils.ReadPtrWithLengthHeader(ref sourceKey.ptr, ref sourceKey.length, ref ptr, recvBufferPtr + bytesRead))
+                    return false;
+
+                // Get the destination key
+                if (!RespReadUtils.ReadPtrWithLengthHeader(ref destinationKey.ptr, ref destinationKey.length, ref ptr, recvBufferPtr + bytesRead))
+                    return false;
+
+                // Get the member to move
+                if (!RespReadUtils.ReadPtrWithLengthHeader(ref sourceMember.ptr, ref sourceMember.length, ref ptr, recvBufferPtr + bytesRead))
+                    return false;
+
+                var keys = new ArgSlice[2] { sourceKey, destinationKey };
+
+                if (NetworkKeyArraySlotVerify(ref keys, false))
+                {
+                    var bufSpan = new ReadOnlySpan<byte>(recvBufferPtr, bytesRead);
+                    if (!DrainCommands(bufSpan, count)) return false;
+                    return true;
+                }
+
+                var status = storageApi.SetMove(sourceKey, destinationKey, sourceMember, out var output);
+
+                if (status == GarnetStatus.NOTFOUND)
```

Comment:
So as it is, we're never returning GarnetStatus.NOTFOUND. What you can do is in case the sourcekey doesn't exist, return  NOTFOUND and write 0 to the output as that's what's expected (CmdStrings.RESP_RETURN_VAL_0).